kratos/tests/cpp_tests/geometries/test_line_3d_3.cpp

```diff
@@ -0,0 +1,335 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "testing/testing.h"
+#include "geometries/line_3d_3.h"
+#include "tests/cpp_tests/geometries/test_geometry.h"
+#include "tests/cpp_tests/geometries/test_shape_function_derivatives.h"
+#include "tests/cpp_tests/geometries/cross_check_shape_functions_values.h"
+
+namespace Kratos {
+namespace Testing {
+
+    /// Factory functions
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsUnitXDirectionLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.0, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsUnitYDirectionLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.0, 1.0, 0.0),
+        Kratos::make_shared<Point>(0.0, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsDiagonalLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 1.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GeneratePointsParabolaLine3D3() {
+        return Kratos::make_shared<Line3D3<Point>>(
+        Kratos::make_shared<Point>(0.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(1.0, 0.0, 0.0),
+        Kratos::make_shared<Point>(0.5, 0.5, 0.0)
+        );
+    }
+
+    /** Generates a point type sample Line3D3N.
+    * @return  Pointer to a Line3D3N
+    */
+    Line3D3<Point>::Pointer GenerateLine3D3WithPoints(Point::Pointer pPointOne, Point::Pointer pPointTwo, Point::Pointer pPointThree ) {
+        return Kratos::make_shared<Line3D3<Point>>(pPointOne, pPointTwo, pPointThree);
+    }
```

Comment:
I'd put all these in an anonymous namespace to avoid exposing them in the complete `Testing` one.

Reply:
I am doing the same as other tests, should I change those too?

Reply:
I don't know by heart what is being done in other tests, but I don't think it makes sense to expose these functions in the complete testing name space. We can discuss about the other tests in a separate PR/issue.

Reply:
I already added the namespace. In any case I am doing a last correction, this is not 100% ready yet

Reply:
Now is ready @rubenzorrilla